### PR TITLE
Add metric for number of unschedulable transactions

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Knowing if there are a lot of unschedulable transactions is useful

#### Summary of Changes
Add a metric

Fixes #

### CODE REVIEW

File: core/src/banking_stage/transaction_scheduler/scheduler_controller.rs [hunk 1]
```diff
@@ -238,6 +245,8 @@ struct SchedulerCountMetrics {
 
     /// Number of transactions scheduled.
     num_scheduled: usize,
+    /// Number of transactions that were unschedulable.
+    num_unschedulable: usize,
```
Comment (taozhu-chicago) [inline core/src/banking_stage/transaction_scheduler/scheduler_controller.rs:249]: Just learn more about these counters, what's the relationship between `num_received`, `num_buffered` and `num_scheduled` `num_unschedulable`? Was thinking `num_unschedulable = num_buffered - num_scheduled`, but guess that isn't correct as one tx could be unschedulable more than once?

File: core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs [hunk 2]
```diff
@@ -109,6 +110,7 @@ impl PrioGraphScheduler {
                 if !blocking_locks.check_locks(transaction.message()) {
                     blocking_locks.take_locks(transaction.message());
                     unschedulable_ids.push(id);
+                    num_unschedulable += 1;
```
Comment (taozhu-chicago) [inline core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs:113]: nit: saturating_add ?

File: core/src/banking_stage/transaction_scheduler/scheduler_controller.rs (hunk 1 above)
Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/scheduler_controller.rs:249]: No explicit relation maintained between the different categories of metrics.
Transactions may be received and buffered in a different interval than they are scheduled in.
As you stated, transactions could also be unschedulable multiple times.

File: core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs (hunk 2 above)
Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs]: done as part of [a20645a](https://github.com/solana-labs/solana/pull/34230/commits/a20645a5e20046e31df8d9973cd18294a37c3516)

I just replaced all scheduler += with saturating_add_assign in that PR.

**Review by taozhu-chicago — APPROVED** [review]
lgtm

**Pull request merged**